pkg/sidecar-controller/snapshot_controller_base.go

```diff
@@ -94,8 +95,15 @@ func NewCSISnapshotSideCarController(
 
 	volumeSnapshotContentInformer.Informer().AddEventHandlerWithResyncPeriod(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc:    func(obj interface{}) { ctrl.enqueueContentWork(obj) },
-			UpdateFunc: func(oldObj, newObj interface{}) { ctrl.enqueueContentWork(newObj) },
+			AddFunc: func(obj interface{}) { ctrl.enqueueContentWork(obj) },
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				newSnapContent := newObj.(*crdv1.VolumeSnapshotContent)
+				oldSnapContent := oldObj.(*crdv1.VolumeSnapshotContent)
+				if newSnapContent.ObjectMeta.Annotations[utils.AnnVolumeSnapshotBeingCreated] != oldSnapContent.ObjectMeta.Annotations[utils.AnnVolumeSnapshotBeingCreated] {
```

Comment:
We also need to make sure this is the only difference.

Reply:
will update different fields at the same time? I means update diferent fileds only do once update or patch.  Beause of the client-go is the fifo queue, I think we no need to keep other are same.

Reply:
Can you add comments to the code to explain why this change is needed?

Reply:
yes

Reply:
We should check the annotation is on the newSnapContent, but not on the oldSnapContent.

Reply:
Only check the "Annotations[utils.AnnVolumeSnapshotBeingCreated]" whether it is empty, no need to compare?

Reply:
But I think whatever remove or add the "Annotations[utils.AnnVolumeSnapshotBeingCreated]"  to the annotations, no need to re-queue.